### PR TITLE
Per project ext updates

### PR DESCRIPTION
**Pull request by shaddockh**
This PR adds a vanilla JavaScript method for creating a per-project extension and also cleans up some lint errors in the TS plugin example.

### CODE REVIEW

**Comment by JoshEngebretson** [comment]
Hey @shaddockh, cool beans thanks for keeping JS in parity with  the editor extension work.
 
I landed #64 which has the updates to the plugin example from @mattbenic you suggested.

It looks like this PR now has some conflicts, it looks mainly to be in some renames for TS and JS clarity

File: EditorPluginExample/Resources/EditorData/TSExample.plugin.ts [hunk 1]
```diff
@@ -115,12 +131,13 @@ class ExamplePluginService implements Editor.HostExtensions.HostEditorService, E
...
 
-    }
+    };
+
 }
 
-const examplePluginService = new ExamplePluginService();
-export default examplePluginService;
\ No newline at end of file
+export default new TSExamplePluginService();
```
Comment (shaddockh) [inline EditorPluginExample/Resources/EditorData/TSExample.plugin.ts]: Discovered that this is a cleaner way of exporting the TS plugin.

**Comment by shaddockh** [comment]
Ok, I've fixed the merge conflicts and migrated over the updates from the TS plugin.

@mattbenic @JoshEngebretson I've also made some tweaks to the existing TS plugin if you want to look them over.  Basically the main changes are
* modified the export line at the end
* display an info box with the name of the item right+clicked in the project / hierarchy menus
* added a scene so that you can test out the hierarchy menu
* duplicated all of the TS functionality into a vanilla JS plugin

**Comment by mattbenic** [comment]
Nicely done :)

**Comment by JoshEngebretson** [comment]
Rocking, looks great! Sorry for the lag, this PR managed to hide on me :)  LANDING!

**Pull request merged**